pkg/k8s/watchers/service.go

```diff
@@ -209,58 +209,62 @@ func (k *K8sServiceWatcher) delK8sSVCs(svc k8s.ServiceID, svcInfo *k8s.Service)
 		return
 	}
 
-	// TODO: cleanup related service with protocol NONE
-
 	scopedLog := log.WithFields(logrus.Fields{
 		logfields.K8sSvcName:   svc.Name,
 		logfields.K8sNamespace: svc.Namespace,
 	})
 
-	repPorts := svcInfo.UniquePorts()
+	for _, checkProtocol := range []bool{true, false} {
```

Comment:
at one point you may not need to do this as all services will be migrated so you may want to use the feature flag?

Reply:
I think putting this behind a feature flag would prevent deleting "old" (ANY) services after an upgrade.

Here's an example:
* user is running v1.15, so no support for service proto differentiation, and has a service deployed
* user upgrades to a Cilium version that supports service proto differentiation, and enables it (or better, doesn't explicitly disable it) 
* after the upgrade, the existing service is maintained as ANY
* at some point the user deletes the service

if this logic was put behind the feature flag, we would never be able to delete that old ANY service from Cilium state/maps, as we would only try to delete the service with the protocol set

Reply:
I worded it wrong, my point is that you only run this if `enable-service-protocol-differentiation` is set , that is the feature flag ... if in a long future there is no need for setting `enable-service-protocol-differentiation` because all cilium clusters are known to have been migrated, you no longer need to do this ... is just a comment

Reply:
ah sorry, I see what you mean now, yes, I need to add also some comments to guide us through all the code paths that need be removed once/if `enable-service-protocol-differentiation` will be deprecated :+1: